web/js/tasks.js

```diff
@@ -884,6 +884,37 @@ Ext.onReady(function(){
             window.scrollTo(0, taskPanel.getEl().getY());
     }
 
+    window.setInterval(function () {
+        if( unsavedChanges == true ) {
+            var panels = tasksScrollArea.items;
+            var valids = true;
+            for(var panel=0; panel<panels.getCount(); panel++) {
+                console.log();
+                if (!panels.get(panel).initTimeField.isValid() ||
+                    !panels.get(panel).endTimeField.isValid() ||
+                    !panels.get(panel).customerComboBox.getRawValue() ||
+                    !panels.get(panel).projectComboBox.getRawValue() ||
+                    !panels.get(panel).storyField.getRawValue() ||
+                    !panels.get(panel).descriptionTextArea.getRawValue()
+                ) {
+                    valids = false;
+                    break;
+                }
+            }
+            if (valids) {
+                myStore.each(function(r) {
+                    if (r.data['story'] != undefined)
+                        r.data['story'] = xmlencode(r.data['story']);
+                    if (r.data['text'] != undefined)
+                        r.data['text'] = xmlencode(r.data['text']);
+                });
+                myStore.save();
+                unsavedChanges = false;
+                Ext.getCmp('status_display').setText("Status: Auto saved at "+ new Date());
```

Comment:
It's not really important if the tasks were manually or automatically saved. You may move this message to the save event in the Store object.